engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/Graph.scala

```diff
@@ -24,11 +27,26 @@ sealed class Graph extends Serializable {
     val copy = new Graph
     copy.rootScope     = this.rootScope.deepCopy(scope_mapping)
     copy.links         = this.links
+    copy.sourceLinks   = this.sourceLinks
+    copy.targetLinks   = this.targetLinks
     copy.globalSymbols = this.globalSymbols
     copy.nextIdCounter = this.nextIdCounter
     copy
   }
 
+  def initLinks(links: Set[Graph.Link]): Unit = {
+    links.foreach { link =>
+      sourceLinks = sourceLinks.updatedWith(link.source)(v =>
+        v.map(s => s + link).orElse(Some(Set(link)))
```

Comment:
According to [scaladoc](https://www.scala-lang.org/api/2.13.7/scala/collection/immutable/Map.html#updatedWith[V1%3E:V](key:K)(remappingFunction:Option[V]=%3EOption[V1]):CC[K,V1]) this is going to add to the `Set` associated with `link.source`. _Just a note_ to be able to read what's happening.